lib/uri_checker.rb

```diff
@@ -0,0 +1,213 @@
+require "uri"
+
+module UriChecker
+  class Report
+    attr_reader :errors, :warnings
+
+    def initialize(errors: nil, warnings: nil)
+      @errors = errors || Hash.new { |hash, key| hash[key] = [] }
+      @warnings = warnings || Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def merge!(other)
```

Comment:
Hmm, I always thought of bang methods as when they are modifying the class itself.